xroot/Overlay/FloatingMenu.lua

```diff
@@ -0,0 +1,177 @@
+end
+
+local function close()
+  local Application = require "Application"
+  Application.setDispatcher()
+  if timer then
+    Timer.cancel(timer)
+    timer = nil
+  end
+
+  local choice
+  if menuArc:hasParent() then
+    Manager.removeMainGraphic(menuArc)
+    choice = menuArc:selectedText()
+    menuArc:clear()
+    fireMenuSelection(choice)
+  end
+
+  target = nil
+  return choice
+end
+
+local function ignoreMainRelease(i)
+  local Application = require "Application"
+  if i == 1 then
+    Application.ignoreEvent(app.EVENT_RELEASE_MAIN1)
+  elseif i == 2 then
+    Application.ignoreEvent(app.EVENT_RELEASE_MAIN2)
+  elseif i == 3 then
+    Application.ignoreEvent(app.EVENT_RELEASE_MAIN3)
+  elseif i == 4 then
+    Application.ignoreEvent(app.EVENT_RELEASE_MAIN4)
+  elseif i == 5 then
+    Application.ignoreEvent(app.EVENT_RELEASE_MAIN5)
+  elseif i == 6 then
+    Application.ignoreEvent(app.EVENT_RELEASE_MAIN6)
+  end
+end
+
+local function dispatcher(event)
+  -- app.logInfo("Overlay.FloatingMenu.dispatcher(%s)",event)
+  local Application = require "Application"
+  local m = Application.getLastMainButtonPressed()
+  if event == app.EVENT_KNOB then
+    local p = app.Encoder_getChange()
+    if p ~= 0 then
+      local Env = require "Env"
+      local threshold = Env.EncoderThreshold.Default
+      local changed = menuArc:encoder(p, false, threshold)
+      if changed then fireMenuChange(menuArc:selectedText()) end
+    end
+  elseif event == app.EVENT_RELEASE_MAIN1 and m == 1 then
+    Application.notify("mainReleased", 1, false)
+  elseif event == app.EVENT_RELEASE_MAIN2 and m == 2 then
+    Application.notify("mainReleased", 2, false)
+  elseif event == app.EVENT_RELEASE_MAIN3 and m == 3 then
+    Application.notify("mainReleased", 3, false)
+  elseif event == app.EVENT_RELEASE_MAIN4 and m == 4 then
+    Application.notify("mainReleased", 4, false)
+  elseif event == app.EVENT_RELEASE_MAIN5 and m == 5 then
+    Application.notify("mainReleased", 5, false)
+  elseif event == app.EVENT_RELEASE_MAIN6 and m == 6 then
+    Application.notify("mainReleased", 6, false)
+  elseif event == app.EVENT_MODE then
+    close()
+    Application.defaultDispatcher(event)
+    ignoreMainRelease(m)
+  elseif event == app.EVENT_STORAGE then
+    close()
+    Application.defaultDispatcher(event)
+    ignoreMainRelease(m)
+  end
+end
+
+local function onOpen()
+  timer = nil
+  local choices, defaultIndex = getMenuItemsAndDefault()
+
+  if #choices > 1 then
+    for _, choice in ipairs(choices) do menuArc:add(choice) end
+    menuArc:select(defaultIndex - 1)
+    Manager.addMainGraphic(menuArc)
+  end
+
+  local Application = require "Application"
+  Application.setDispatcher(dispatcher)
+
+  fireMenuEnter()
+end
+
+local function getDelayTime()
+  local Settings = require "Settings"
+  local choice = Settings.get("floatingMenuDelay")
+  local values = {
+    ["0.1s"] = 0.1,
+    ["0.2s"] = 0.2,
+    ["0.3s"] = 0.3,
+    ["0.5s"] = 0.5,
```

Comment:
Ha well I meant 50ms instead of 500ms :p

Reply:
Oops

Reply:
Now 50ms, 0.1s, 0.2s, 0.4s.

50ms is really short for a long press! :thinking: 

[er-301-v0.6.16-testing.zip](https://github.com/odevices/er-301/files/7109165/er-301-v0.6.16-testing.zip)

Reply:
Ha yea I'm running it now and 0.1s is also proving to be _quite_ short.. trying the others now

Reply:
Ok the difference between 50 and 100ms is basically not noticeable, how about this:

100ms - Very short, practically instant
300ms - Normal, almost always intentional
500ms - Slow, extremely intentional

That's my take away at least. I'm realizing that 0.3s was probably the best choice in the first place so.. my bad ha.

Reply:
That matches my experience also.